helmclient.go

```diff
@@ -67,13 +68,28 @@ func New(config Config) (*Client, error) {
 
 // DeleteRelease uninstalls a chart given its release name.
 func (c *Client) DeleteRelease(releaseName string, options ...helmclient.DeleteOption) error {
-	t, err := c.newTunnel()
-	if err != nil {
-		return microerror.Mask(err)
+	o := func() error {
+		t, err := c.newTunnel()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		defer c.closeTunnel(t)
+
+		_, err = c.newHelmClientFromTunnel(t).DeleteRelease(releaseName, options...)
+		if IsReleaseNotFound(err) {
+			return backoff.Permanent(releaseNotFoundError)
```

Comment:
So note that we have to use a permanent error here to tell the retry to stop since we actually only want to catch unexpected stuff like broken network connections. Here we cannot use error masking because of the backoff internals. When the backoff stops it unwraps the given error and we can use masking again.